### PR TITLE
[igemm]V4r1 bwd xdlops

### PR DESCRIPTION
**Pull request by asleepzzz**
this kernel can support 256x128 block size and improve 3x3 performance

### CODE REVIEW

**Comment by asleepzzz** [comment]
[V4R1_perf.xlsx](https://github.com/ROCmSoftwarePlatform/MIOpen/files/5182843/V4R1_perf.xlsx)

**Comment by atamazov** [comment]
@asleepzzz > [V4R1_perf.xlsx](https://github.com/ROCmSoftwarePlatform/MIOpen/files/5182843/V4R1_perf.xlsx)

- (1) Could you please attach Find logs (taken with MIOPEN_LOG_LEVEL=5 and `-F 2`) for the cases where this new kernel wins?
- (2) The .xls file shows only 7 configs. Have you tested more?
- (3) The title of PR says that kernel is ___V4R1___. Excel table shows ConvHipImplicitGemmBwdData<b><i>V1R1</i></b>Xdlops. Why?

**Comment by atamazov** [comment]
@asleepzzz 
- (4) Does this PR require retuning of certain PerformanceConfgs?
- (5) How this kernel was tested for correctness?
- (6) Any regressions, i.e. cases when old kernel is faster than this one?
- (7) Does this change affect tuning time of this Solver?

File: src/include/miopen/solver.hpp [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
```
Comment (shaojiewang) [inline src/include/miopen/solver.hpp:25]: Could you please remove the blank space at the beginning of each line?

**Comment by asleepzzz** [comment]
> @asleepzzz > [V4R1_perf.xlsx](https://github.com/ROCmSoftwarePlatform/MIOpen/files/5182843/V4R1_perf.xlsx)
> 
> * (1) Could you please attach Find logs (taken with MIOPEN_LOG_LEVEL=5 and `-F 2`) for the cases when this kernel wins, prior and after this change (I see 3 such cases in .xlsx)?
> * (2) The .xlsx file shows only 7 configs. Have you tested more?
> * (3) The title of PR says that kernel is _**V4R1**_. Excel table shows ConvHipImplicitGemmBwdDataV1R1Xdlops. Why?

(2)this kernel is for non 1x1 configs,I also test 1x1 configs, not worse than V1R1 
(3)before this PR, MIOPEN choose V1R1

**Comment by asleepzzz** [comment]
> @asleepzzz
> 
> * (4) Does this PR require retuning of certain PerformanceConfgs?
> * (5) How this kernel was tested for correctness?
> * (6) Any regressions, i.e. cases when old kernel is faster than this one?
> * (7) Does this change affect tuning time of this Solver?

(4)yes,this kernel is not dynamic
(5)pass the MIOpen verification

**Comment by asleepzzz** [comment]
[logs_old_v4r1.txt](https://github.com/ROCmSoftwarePlatform/MIOpen/files/5194223/logs_old_v4r1.txt)
[logs_v4r1.txt](https://github.com/ROCmSoftwarePlatform/MIOpen/files/5194225/logs_v4r1.txt)

**Comment by asleepzzz** [comment]
for ./bin/MIOpenDriver convfp16 -F 2 -n 128 -c 128 -H 58 -W 58 -k 128 -y 3 -x 3 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -g 1 -t 1[
old version tuning time: 371sec
new version tuning time: 1114 sec
configs become more,but still much less than other solver

**Comment by asleepzzz** [comment]
[V4R1_perf (1).xlsx](https://github.com/ROCmSoftwarePlatform/MIOpen/files/5199186/V4R1_perf.1.xlsx)

Even 1x1 stide1,V4R1 is better than V1R1 ,maybe we can change order of solver list......

File: src/solver.cpp [hunk 2]
```diff
@@ -296,6 +296,9 @@ inline SolverRegistrar::SolverRegistrar(IdRegistryData& registry)
     RegisterWithSolver(
         registry, ++id, ConvHipImplicitGemmBwdDataV4R1{}, miopenConvolutionAlgoImplicitGEMM);
 
+    RegisterWithSolver(
```
Comment (atamazov) [inline src/solver.cpp:299]: This is wrong. Solvers should be always appended to the end of the registry.

File: src/solver.cpp [hunk 3]
```diff
@@ -296,6 +296,9 @@ inline SolverRegistrar::SolverRegistrar(IdRegistryData& registry)
...
+        registry, ++id, ConvHipImplicitGemmBwdDataV4R1Xdlops{}, miopenConvolutionAlgoImplicitGEMM);
```
Comment (atamazov) [inline src/solver.cpp:300]: [Notice] If you would like to change the order ("priority"), please modify the list in the mlo_dir_conv.cpp.

**Review by atamazov — CHANGES_REQUESTED** [review]

File: src/solver.cpp (hunk 2 above)
Comment (asleepzzz) [reply src/solver.cpp:299]: I will collect inception data to determine change order or not

File: src/solver.cpp (hunk 3 above)
Comment (asleepzzz) [reply src/solver.cpp:300]: I will collect inception data to determine change order or not

File: src/solver.cpp [hunk 4]
```diff
@@ -296,6 +296,9 @@ inline SolverRegistrar::SolverRegistrar(IdRegistryData& registry)
...
+
```
[the remark this answers is not in the document]
Comment (asleepzzz) [reply src/solver.cpp:301]: db need to run again

**Comment by asleepzzz** [comment]
for some inception configs, v1r1 better than v4r1 ,so don't change order

File: src/solver.cpp (hunk 2 above)
Comment (atamazov) [reply src/solver.cpp]: ?

**Comment by atamazov** [comment]
@asleepzzz Please do not mark my review comments as Resolved. I will do that myself. Thanks.

File: src/solver.cpp (hunk 2 above)
Comment (asleepzzz) [reply src/solver.cpp]: the latest commit do not change order

File: src/solver.cpp (hunk 3 above)
Comment (asleepzzz) [reply src/solver.cpp]: the latest commit do not change order

File: src/solver.cpp (hunk 2 above)
Comment (asroy) [reply src/solver.cpp]: @asleepzzz This is to register a solver, by giving it an unique ID, this ID will be used in find-db and perf-db etc. A new solver should always put at the very end of the list, so it doesn't change the IDs of existing solvers.

The order of the solver here will not affect which solver will be picked.

File: src/solver.cpp (hunk 3 above)
Comment (asroy) [reply src/solver.cpp]: @asleepzzz This is to register a solver, by giving it an unique ID, this ID will be used in find-db and perf-db etc. A new solver should always put at the very end of the list, so it doesn't change the IDs of existing solvers.

The order of the solver here will not affect which solver will be picked.

File: src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp [hunk 5]
```diff
@@ -332,6 +367,151 @@ bool PerformanceImplicitGemmBwdDataV4R1Xdlops::IsValid(const ConvolutionContext&
...
+    const ConvolutionContext& ctx) const
+{
+    // somehow, 128x128 wave-wise GEMM tend to spill register
+    // TODO revisit this when 128x128 wave-wise GEMM become efficient
+    {
+        if(GemmMPerWave * GemmNPerWave > 64 * 128)
+            return false;
+    }
+
+#if WORKAROUND_SWDEV_240356
```
Comment (asroy) [inline src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp:380]: The issue is resolver by #403. This workaround can be removed

File: src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp [hunk 6]
```diff
@@ -332,6 +367,151 @@ bool PerformanceImplicitGemmBwdDataV4R1Xdlops::IsValid(const ConvolutionContext&
...
+
+        std::tie(std::ignore, gemm_m, gemm_n, std::ignore) =
+            ConvHipImplicitGemmBwdDataV4R1Xdlops::CalculateGemmSize(ctx, 0);
+
+        // this is grid size using current blockwise-GEMM
+        const int grid_size = (gemm_m * gemm_n) / (GemmMPerBlock * GemmNPerBlock);
+
+        // this the the biggest blockwise-GEMM you can do
+        int max_blockwise_gemm_size =
+#if WORKAROUND_SWDEV_240356
```
Comment (asroy) [inline src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp:400]: The issue is resolver by #403. This workaround can be removed

File: src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp [hunk 7]
```diff
@@ -332,6 +367,151 @@ bool PerformanceImplicitGemmBwdDataV4R1Xdlops::IsValid(const ConvolutionContext&
...
+    // don't need too many waves per block
+    {
+        const int wave_per_block = (GemmMPerBlock / GemmMPerWave) * (GemmNPerBlock / GemmNPerWave);
+
+        if(!(wave_per_block > 1 && wave_per_block <= 4))
+        {
+            return false;
+        }
+    }
+    /*3x3 should use these configs
```
Comment (asroy) [inline src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp:448]: These logic are better for 3x3?

File: src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp [hunk 8]
```diff
@@ -154,38 +184,56 @@ PerformanceImplicitGemmBwdDataV4R1Xdlops::CalculateGemmBBlockCopyPerformancePara
...
         // calculate threadwise copy size
-        int b_data_per_thread_copy = (GemmKPerBlock * GemmNPerBlock) / BlockSize;
+        int b_data_per_thread_copy = (GemmKPerBlock * GemmNPerBlock * GemmKPACKSize) / BlockSize;
 
         if(!(b_data_per_thread_copy > 0))
             MIOPEN_THROW("invalid performance parameter");
 
         // GemmBBlockCopySrcDataPerRead_GemmN also bounded by size of threadwise copy
-        SrcDataPerRead_GemmN = gcd(SrcDataPerRead_GemmN, b_data_per_thread_copy);
+        SrcDataPerRead_GemmN                  = gcd(SrcDataPerRead_GemmN, b_data_per_thread_copy);
```
Comment (asroy) [inline src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp:193]: GemmN = merge (N, HTildaSlice, WTildaSlice). But here  the calculation of ```SrcDataPerRead_GemmN``` doesn't consider what is the value of  ```WTildaSlice```. If ```WTildaSlice``` is odd number, then we cannot do vector read on GemmN.

I'm surprised this logic has not caused any failure. We need to discuss on this logic, it seem wrong to me

**Review by asroy — CHANGES_REQUESTED** [review]

File: src/kernels/composable_kernel/include/kernel_algorithm/gridwise_convolution_backward_data_implicit_gemm_v4r1_xdlops_fp16_bfp16_gnchw_gkcyx_gnkhw.hpp [hunk 9]
```diff
@@ -468,41 +466,43 @@ struct GridwiseConvolutionBackwardDataImplicitGemm_v4r1_xdlops_fp16_bfp16_gnchw_
...
+            decltype(out_gemmg_gemmk_gemmn_gemmkpack_global_desc),
+            decltype(in_gemmg_gemmm_gemmn_global_desc),
+            GemmMPerBlock,
+            GemmNPerBlock,
+            GemmKPerBlock,
+            GemmMPerWave,
+            GemmNPerWave,
+            GemmABlockCopyThreadSliceLengths_GemmG_GemmK_GemmM_GemmKPACK,
+            GemmABlockCopyThreadClusterLengths_GemmG_GemmK_GemmM_GemmKPACK,
+            Sequence<0, 1, 2, 3>,
```
[the remark this answers is not in the document]
Comment (asleepzzz) [reply src/kernels/composable_kernel/include/kernel_algorithm/gridwise_convolution_backward_data_implicit_gemm_v4r1_xdlops_fp16_bfp16_gnchw_gkcyx_gnkhw.hpp:485]: give the best performance

File: src/kernels/composable_kernel/include/kernel_algorithm/gridwise_convolution_backward_data_implicit_gemm_v4r1_xdlops_fp16_bfp16_gnchw_gkcyx_gnkhw.hpp [hunk 10]
```diff
@@ -468,41 +466,43 @@ struct GridwiseConvolutionBackwardDataImplicitGemm_v4r1_xdlops_fp16_bfp16_gnchw_
...
+            Sequence<0, 1, 2, 3>,
+            Sequence<0, 1, 2, 3>,
+            2,
+            GemmABlockCopySrcDataPerRead_GemmM,
+            GemmABlockCopyDstDataPerWrite_GemmKPACK,
+            GemmBBlockCopyThreadSliceLengths_GemmG_GemmK_GemmN_GemmKPACK,
+            GemmBBlockCopyThreadClusterLengths_GemmG_GemmK_GemmN_GemmKPACK,
+            Sequence<0, 1, 2, 3>,
```
[the remark this answers is not in the document]
Comment (asleepzzz) [reply src/kernels/composable_kernel/include/kernel_algorithm/gridwise_convolution_backward_data_implicit_gemm_v4r1_xdlops_fp16_bfp16_gnchw_gkcyx_gnkhw.hpp:493]: give the best performance

File: src/kernels/composable_kernel/include/kernel_algorithm/gridwise_convolution_backward_data_implicit_gemm_v4r1_xdlops_fp16_bfp16_gnchw_gkcyx_gnkhw.hpp [hunk 11]
```diff
@@ -468,41 +466,43 @@ struct GridwiseConvolutionBackwardDataImplicitGemm_v4r1_xdlops_fp16_bfp16_gnchw_
...
+            Sequence<0, 1, 2, 3>,
+            Sequence<0, 1, 2, 3>,
+            2,
+            GemmBBlockCopySrcDataPerRead_GemmN,
+            GemmBBlockCopyDstDataPerWrite_GemmKPACK,
+            InMemoryDataOperation::Set,
+#if MIOPEN_USE_FP16 || MIOPEN_USE_BFP16
+            NBlock1MBlock0
```
[the remark this answers is not in the document]
Comment (asleepzzz) [reply src/kernels/composable_kernel/include/kernel_algorithm/gridwise_convolution_backward_data_implicit_gemm_v4r1_xdlops_fp16_bfp16_gnchw_gkcyx_gnkhw.hpp:501]: Yes, fp16 should use NBlock1MBlock0

**Comment by atamazov** [comment]
@asleepzzz 

>> * (1) Could you please attach Find logs (taken with MIOPEN_LOG_LEVEL=5 and `-F 2`) for the cases when this kernel wins, prior and after this change (I see 3 such cases in .xlsx)?

Please answer.

>> * (4) Does this PR require retuning of certain PerformanceConfgs?
>
> (4)yes, this kernel is not dynamic

"Dynamic" and "Tunable" are orthogonal notions.

I know that original Solver needs tuning. The question is about ** _ retuning _ **. Your PR changes existing Solver, so optimal values ​​of its PerformanceConfig may change. Or, format of records in performance database may change. If this is so, then we have to clearly inform Tuna guys.

If retuning is required, please notice that in the PR description and set appropriate label. Thanks.

>> * (5) How this kernel was tested for correctness?
>
> (5)pass the MIOpen verification

What is "MIOpen verification"? How can I reproduce these tests?

>> * (6) Any regressions, i.e. cases when old kernel is faster than this one?

Please answer.

>> * (7) Does this change affect tuning time of this Solver?

Please answer.

**Review by asleepzzz — COMMENTED** [review]
> @asleepzzz
> 
> > > * (1) Could you please attach Find logs (taken with MIOPEN_LOG_LEVEL=5 and `-F 2`) for the cases when this kernel wins, prior and after this change (I see 3 such cases in .xlsx)?
> 
> OK, I'll attach more
> 
> > > * (4) Does this PR require retuning of certain PerformanceConfgs?
> > 
> > yes
> 
> > > * (5) How this kernel was tested for correctness?
> > 
> > I use MIOpenDriver to run resnet and inception configs, It showed Convolution Verifies on CPU and GPU

> > > * (6) Any regressions, i.e. cases when old kernel is faster than this one?
> 
> I need to coolect more data,please wait
> 
> > > * (7) Does this change affect tuning time of this Solver?
> 
> Please answer.

**Comment by asleepzzz** [comment]
[logs_old_v4r1.txt](https://github.com/ROCmSoftwarePlatform/MIOpen/files/5265306/logs_old_v4r1.txt)
[logs_new_v4r1.txt](https://github.com/ROCmSoftwarePlatform/MIOpen/files/5265307/logs_new_v4r1.txt)
Hi atamazov , these logs are old and new solver
average of tuning paremater: old is 9 ,new is 62
tuning time will increase due to we add more block size, we can use 256x128 now ,so we can better performance

for some cases,I don't know this called regression or not,most of them drop under 0.1%......
 
MIOpenDriver convfp16 -n 256 -c 512 -H 28 -W 28 -k 1024 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -F 2 -t 1
old:1.205ms
new:1.206ms 
drop 0.08%

MIOpenDriver convfp16 -n 256 -c 512 -H 7 -W 7 -k 512 -y 3 -x 3 -p 1 -q 1 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -F 2 -t 1
old:1.341ms
new:1.342ms
drop 0.07%

MIOpenDriver convfp16 -n 256 -c 64 -H 56 -W 56 -k 64 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -F 2 -t 1
old:0.2187ms
new:0.2190ms
drop 0.1%

MIOpenDriver convfp16 -n 256 -c 64 -H 56 -W 56 -k 64 -y 3 -x 3 -p 1 -q 1 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -F 2 -t 1
old:1.1989ms
new:1.199ms
drop 0.008%

File: src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp [hunk 12]
```diff
@@ -353,13 +535,17 @@ PerformanceImplicitGemmBwdDataV4R1Xdlops::PerformanceImplicitGemmBwdDataV4R1Xdlo
...
     bool use_spare_set_)
     : GemmNPerBlock(GemmNPerBlock_),
       GemmMPerBlock(GemmMPerBlock_),
       GemmKPerBlock(GemmKPerBlock_),
       GemmKPACKSize(GemmKPACKSize_),
       GemmMPerWave(GemmMPerWave_),
       GemmNPerWave(GemmNPerWave_),
+      GemmAThreadCopyMoreGemmK(GemmAThreadCopyMoreGemmK_),
+      GemmBThreadCopyMoreGemmKPack(GemmBThreadCopyMoreGemmKPack_),
       use_spare_set(use_spare_set_)
```
[the remark this answers is not in the document]
Comment (asleepzzz) [reply src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp]: already remove

File: src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp (hunk 7 above)
Comment (asleepzzz) [reply src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp:455]: I remove 3x3 configs due to this logic is from fwd and I test it,
not suitable to bwd, without these configs ,we can't get better performance

File: src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp (hunk 8 above)
Comment (asleepzzz) [reply src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp:200]: Yes ,we can dicusss it tomorrow

File: src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp (hunk 7 above)
Comment (asroy) [reply src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp:456]: please remove the commented code

**Review by asroy — DISMISSED** [review]
LGTM. Just remember to remove the commented code

**Comment by asleepzzz** [comment]
> LGTM. Just remember to remove the commented code

done

**Review by asroy — APPROVED** [review]

**Comment by atamazov** [comment]
@asleepzzz Thanks for the logs. I am preparing some analysis now...

**Comment by atamazov** [comment]
> for some cases,I don't know this called regression or not,most of them drop under 0.1%...

This doesn't matter. Most likely this is due to jitter. Usually I ignore diffs < 1%. Diffs >5% worth close consideration.

**Comment by atamazov** [comment]
## Some questions about logs

> logs_old_v4r1.txt
> logs_new_v4r1.txt

Were these logs obtained with `MIOPEN_DEBUG_IMPLICIT_GEMM_FIND_ALL_SOLUTIONS=1`?

In the logs I see the following:
```
MIOpen(HIP): Warning [Monitor] 1241/0/1437 3.04385, best within recent 21: 4.1085 #1238 1,16,2,32,3,2,2,4, ETA:32.2508 sec.=====  204200
```
Please notice `=====  <decimal number>` at the end. Should I ignore these numbers?

It seems like these logs were collected with some code changes that are not included into this PR, right? What are those changes? Only logging?

## Log analysis.

With `MIOPEN_DEBUG_IMPLICIT_GEMM_FIND_ALL_SOLUTIONS=0`, the PR has no effect. The reason is that V1R1-xdlops precedes V4R1-xdlops in `GetImplicitGemmSolvers`.

With `MIOPEN_DEBUG_IMPLICIT_GEMM_FIND_ALL_SOLUTIONS=1`, V4R1-xdlops wins in 13 configs out of 21 total. The average performance gain is about of 4%. The drawback is ~5 hours of additional tuning time (15 minutes per config on average). Average tuning time slowdown of new V4R1-xdlops vs. old one is ~9X.

Note that it seems ___highly unlikely___ that we will use `MIOPEN_DEBUG_IMPLICIT_GEMM_FIND_ALL_SOLUTIONS=1` in the near future. Please see https://github.com/AMDComputeLibraries/MLOpen/pull/2520 for explanation.

Summary table (`MIOPEN_DEBUG_IMPLICIT_GEMM_FIND_ALL_SOLUTIONS=1`): [v4r1-xdlops-new-summary-table.xlsx](https://github.com/ROCmSoftwarePlatform/MIOpen/files/5279941/v4r1-xdlops-new-summary-table.xlsx)

## Conclusions:

- The PR wouldn't give any performance gain right away. However, I do not see any harm from merging it.
- Moreover, if it fixes #461, then it is definitely worth merging.
- The cost of this PR is necessity of re-tuning.
- I highly recommend considering optimization of V4R1-xdlops tuning time in the future.

**Comment by atamazov** [comment]
@asleepzzz
> > * (5) How this kernel was tested for correctness?
>
> (5)pass the MIOpen verification

Obviously ;) Could you please clarify how many configs have you tested with this Solver, i.e. with MIOPEN_DEBUG_FIND_ONLY_SOLVER=60? (Better, attach a list of that configs)

**Comment by atamazov** [comment]
W.R.T. re-tuning. At first, it is Ok to simply remove all the V4R1-xdlops PerformanceConfigs from the system-perf-db. The tuning "score" is about of ~1.1 on average which means that `HeuristicInit()` gives reasonably good default PerformanceConfigs.

/cc @JehandadKhan @daniellowell

**Review by atamazov — APPROVED** [review]
Good to go provided that it has been thoroughly tested, see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/419#issuecomment-698665702

**Comment by atamazov** [comment]
[off-topic] @asleepzzz :warning: Please ___do not ever___ use editing others' comments for your replies. Basically nobody receive notifications about edits, so we didn't read your response(s). I saw it by accident.

**Comment by atamazov** [comment]
@asleepzzz 
> one log is only use V4R4 with this PR

V4R1, I guess

> another is use other solvers and do not have this PR

Usually I call this "baseline"

> This PR is for3x3 stride2,I will try to reduce tuning time

Good!

**Pull request merged**